include/libcloudph++/common/ice_nucleation.hpp

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include "units.hpp"
+
+namespace libcloudphxx
+{
+  namespace common
+  {
+    namespace ice_nucleation
+    {
+      enum class INP_t {mineral}; // types of ice nucleating particles, TODO: add more types
+
+      // Freezing temperature as defined in Shima et al., 2020
+      template <typename real_t>
+      BOOST_GPU_ENABLED
+      quantity<si::temperature, real_t> T_freeze(
```

Comment:
The idea is to have 'physics' in include/ files, and implementation in src/ files. Therefore it would be better to define T_freeze_CDF here, but to move the sampling logic  to src/